internal/ingest/task.go

```diff
@@ -4,23 +4,47 @@ import (
 	"context"
 	"database/sql"
 	"fmt"
+	"slices"
 	"time"
 
 	goaingest "github.com/artefactual-sdps/enduro/internal/api/gen/ingest"
 	"github.com/artefactual-sdps/enduro/internal/datatypes"
 	"github.com/artefactual-sdps/enduro/internal/enums"
+	"github.com/artefactual-sdps/enduro/internal/persistence"
 )
 
 func (svc *ingestImpl) CreateTask(ctx context.Context, task *datatypes.Task) error {
-	err := svc.perSvc.CreateTask(ctx, task)
-	if err != nil {
+	tasks := []*datatypes.Task{task}
+	return svc.CreateTasks(ctx, persistence.TaskSequence(slices.Values(tasks)))
+}
+
+func (svc *ingestImpl) CreateTasks(
+	ctx context.Context,
+	seq persistence.TaskSequence,
+) error {
+	// Tee the sequence to capture task pointers while forwarding them to the
+	// persistence layer. This allows us to access the tasks (with their
+	// database-generated IDs) after persistence completes, so we can publish
+	// events without consuming the sequence twice or changing the persistence
+	// API.
+	var tasks []*datatypes.Task
```

Comment:
This could probably benefit from pre-allocating the slice with a larger capacity to minimized the number of copies to a new slice when the capacity is exceeded.  Considering the number of tasks passed through here is usually between one and hundred maybe something like `tasks := make([]*datatypes.Task, 0, 100)`?

Reply:
I think you're right!